### ISSUE TITLE
Constants

### ISSUE THREAD

**Issue by Yurlungur** [lanl/singularity-eos#443]
singularity-opac has a constants and unit systems infrastructure. singularity-eos does not. These are inconsistent and also should ideally be shared. I wonder if this infrastructure should in fact live in ports-of-call?

**Comment by jhp-lanl** [comment]
What is the functional difference between a 'unit system' and the units modifier we have?

**Comment by Yurlungur** [comment]
So there are a few independent concepts, that I think are getting confused
1. Physical constants like $k_b$
2. We may wish to pass in quantities in different  unit systems
3. For some tests problems (at least in radiation), its sometimes useful to set physical constants (like the stefan-boltzmann constant) to weird numbers

Concept (2) is completley handled in singularity-eos by the `UnitSystem` modifier. Concepts (1) and (3) are *both* handled in singularity-opac by a compile time constants object where you set constants to be whatever you like.

What I'm wondering about is whether or not we should also add a (1) capability to singularity-eos. We could provide a set of physical constants in cgs.

**Comment by jhp-lanl** [comment]
If you're just talking about adding constants in CGS units, I think that's fine.

I'd also support a discussion about compile time unit systems to better mesh with host codes, but I'd want to learn more about how other codes handle unit system changes. For example, if a code is cg$\mu$ ($\mu$ being microseconds), then I can see how it would be really nice to just avoid unit conversions entirely and have everything self-consistent.

**Comment by Yurlungur** [comment]
Yeah we're on the same page. I don't want to go down the compile time unit system path right now, or at least I don't want to spend time building something as we've got other fish to fry.
